projects/storefrontlib/src/lib/cms/services/component-mapper.service.ts

```diff
@@ -58,4 +73,59 @@ export class ComponentMapperService {
 
     return factoryEntry ? factoryEntry[0] : null;
   }
+
+  isWebComponent(typeCode: string): boolean {
+    return (this.getType(typeCode) || '').includes('#');
+  }
+
+  initWebComponent(
+    componentType: string,
+    renderer: Renderer2
+  ): Promise<string> {
+    return new Promise(resolve => {
+      const [path, selector] = this.getType(componentType).split('#');
+
+      let script = this.loadedWebComponents[path];
+
+      if (!script) {
+        script = renderer.createElement('script');
+        this.loadedWebComponents[path] = script;
+        script.setAttribute('src', path);
+        renderer.appendChild(this.document.body, script);
+
+        if (isPlatformBrowser(this.platform)) {
```

Comment:
Maybe for readability purposes, it would be good to add some comment why we are chaining those `script.onload`s.